### PR TITLE
fix(federation): Remove the notion of @inaccessible from subgraphs

### LINKED ISSUE #867: composeAndValidate() unexpectedly generates invalid GraphQL when using @tag/@inaccessible
`composeAndValidate()` in `@apollo/federation@0.26.0` currently generates a supergraph schema that is invalid GraphQL when given the following subgraph schemas:
- `inventory.graphql`
    ```graphql
    extend type Product @key(fields: "id") {
      id: ID! @external @tag(name: "hi from inventory")
      dimensions: ProductDimension @external
      delivery(zip: String): DeliveryEstimates @requires(fields: "dimensions { size weight }")
    }
    type ProductDimension {
      size: String
      weight: Float @inaccessible
    }
    type DeliveryEstimates {
      estimatedDelivery: String
      fastestDelivery: String
    }
    ```
- `users.graphl`
    ```graphql
    type User @key(fields:"email") {
        email:ID!
        name: String
        totalProductsCreated: Int
    }
    ```
- `products.graphl`
    ```graphql
    type Product @key(fields: "id") @key(fields: "sku package") @key(fields: "sku variation { id }"){
      id: ID!
      sku: String
      package: String
      variation: ProductVariation
      dimensions: ProductDimension
      createdBy: User @provides(fields: "totalProductsCreated")
    }
    type ProductVariation {
      id: ID!
    }
    enum Color {
      BLUE
      GREEN
    }
    type ProductDimension {
      size: String
      weight: Float
    }
    extend type Query {
      allProducts: [Product]
      product(id: ID!): Product
    }
    extend type User @key(fields: "email") {
      email: ID! @external
      totalProductsCreated: Int @external
    }
    ```

The supergraph schema generated is specifically:
```graphql
schema
  @core(feature: "https://specs.apollo.dev/core/v0.1"),
  @core(feature: "https://specs.apollo.dev/join/v0.1"),
  @core(feature: "https://specs.apollo.dev/inaccessible/v0.1")
{
  query: Query
}
directive @core(feature: String!) repeatable on SCHEMA
directive @join__field(graph: join__Graph, requires: join__FieldSet, provides: join__FieldSet) on FIELD_DEFINITION
directive @join__type(graph: join__Graph!, key: join__FieldSet) repeatable on OBJECT | INTERFACE
directive @join__owner(graph: join__Graph!) on OBJECT | INTERFACE
directive @join__graph(name: String!, url: String!) on ENUM_VALUE
directive @inaccessible on FIELD_DEFINITION
enum Color {
  BLUE
  GREEN
}
type DeliveryEstimates {
  estimatedDelivery: String
  fastestDelivery: String
}
scalar join__FieldSet
enum join__Graph {
  INVENTORY @join__graph(name: "inventory" url: "http://inventory:4000/graphql")
  PRODUCTS @join__graph(name: "products" url: "http://product:4000/graphql")
  USERS @join__graph(name: "users" url: "http://users:4000/graphql")
}
type Product
  @join__owner(graph: PRODUCTS)
  @join__type(graph: PRODUCTS, key: "id")
  @join__type(graph: PRODUCTS, key: "sku package")
  @join__type(graph: PRODUCTS, key: "sku variation{id}")
  @join__type(graph: INVENTORY, key: "id")
{
  createdBy: User @join__field(graph: PRODUCTS, provides: "totalProductsCreated")
  delivery(zip: String): DeliveryEstimates @join__field(graph: INVENTORY, requires: "dimensions{size weight}")
  dimensions: ProductDimension @join__field(graph: PRODUCTS)
  id: ID! @join__field(graph: PRODUCTS) @tag(name: "hi from inventory")
  package: String @join__field(graph: PRODUCTS)
  sku: String @join__field(graph: PRODUCTS)
  variation: ProductVariation @join__field(graph: PRODUCTS)
}
type ProductDimension {
  size: String
  weight: Float @inaccessible
}
type ProductVariation {
  id: ID!
}
type Query {
  allProducts: [Product] @join__field(graph: PRODUCTS)
  product(id: ID!): Product @join__field(graph: PRODUCTS)
}
type User
  @join__owner(graph: USERS)
  @join__type(graph: USERS, key: "email")
  @join__type(graph: PRODUCTS, key: "email")
{
  email: ID! @join__field(graph: USERS)
  name: String @join__field(graph: USERS)
  totalProductsCreated: Int @join__field(graph: USERS)
}
```

Which notably doesn’t have a `@tag` directive definition, but does have usages of `@tag`. (So I’m guessing this is a bug with #859 .)

### PR DESCRIPTION
**Pull request by trevor-scheer**
We've decided to go forward with the decision to ignore the usage of @inaccessible at the subgraph level, and instead allow @inaccessible to be a result of using @tag + Studio governance.

This directly relates to #869 but will take precedence due to the updated decision.

This PR also handles the rename of `appliedDirectives` recommended in feedback on #859.

Fixes #867.

### CODE REVIEW

**Review by glasser — APPROVED** [review]
@trevor-scheer says that making progress today on another PR will be much easier if this PR is merged now, and that appropriate reviewers aren't available in this time zone now. Clicking the green button without a real review; @trevor-scheer will find somebody to do post-merge review.

**Comment by pcmanus** [comment]
Had a quick post-merge review look and didn't spot any obvious issue.

One remark that crossed my mind is: what we mostly wanted here was the updates to `compose.ts` so it doesn't handle/propagate `@inaccessible` on subgraphs, but as we kind of want to allow studio to transform `@tag` into `@inacessible` in the supergraph, would it have made sense to preserve the handling of `@inacessible` in `printSupergraphSdl.ts`? I suppose it's somewhat moot though if studio doesn't use `printSupergraphSdl.ts` (does it?).

File: federation-js/src/service/__tests__/printSupergraphSdl.test.ts [hunk 1]
```diff
@@ -311,11 +308,9 @@ describe('printSupergraphSdl', () => {
     `);
   });
 
-  it('prints a fully composed schema without @tag or @inaccessible correctly', () => {
+  it('prints a fully composed schema without @tag orcorrectly', () => {
```
Comment (pcmanus) [inline federation-js/src/service/__tests__/printSupergraphSdl.test.ts:311]: Fyi, you left the `or` mistakenly, but I took the liberty to ninja-merge the fix in https://github.com/apollographql/federation/commit/e9e9a66d239c05123044edc14d79f60013abcf27.

**Comment by abernix** [comment]
> would it have made sense to preserve the handling of `@inacessible` in `printSupergraphSdl.ts`?

Good question! I found an internal thread which helped me answer this.  Preserving the handling may still make sense but _currently_ that functionality is **not** necessitated by the Studio team.  We can re-introduce it if they need it for their use cases but since that is the only current use-case (that I'm aware of) I think it's fine to leave it out.

cc @jsegaran in case you want to propose it be introduced eventually!, but I can also see this need being superseded by a more principled way of directive life-cycle management that we already have in mind.

File: federation-js/src/service/__tests__/printSupergraphSdl.test.ts (hunk 1 above)
Comment (trevor-scheer) [reply federation-js/src/service/__tests__/printSupergraphSdl.test.ts:311]: Thanks!

**Comment by trevor-scheer** [comment]
Thanks for taking a look and your intuition matched mine @pcmanus. As @abernix mentioned, I did end up considering this (after the fact) and fortunately it seems Studio doesn't need that functionality. If we ever do need to reintroduce it though, this PR will make for a great reference of what to re-undo 😄

**Pull request merged**